scripts/lib.mjs

```diff
@@ -0,0 +1,40 @@
+import fsPromises from "fs/promises";
+import crypto from "crypto";
+
+const loadAllPackageLocks = async (root) => {
+	const files = await fsPromises.readdir(root, { withFileTypes: true });
+
+	return await files.reduce(async (locksP, file) => {
+		const fileName = `${root}/${file.name}`;
+		const locks = await locksP;
+
+		if (file.isDirectory())
+			return [...locks, ...(await loadAllPackageLocks(fileName))];
+
+		if (file.name === "package-lock.json")
+			return [...locks, fileName];
+
+		return locks;
+	}, Promise.resolve([]));
+};
+
+const loadJSONFile = (file) => fsPromises.readFile(file, { encoding: 'utf8' }).then(JSON.parse);
+
+const getHashOf = (type, file) => fsPromises.readFile(file).then((contents) => {
+	const hash = crypto.createHash(type);
+	hash.setEncoding("hex");
+	hash.update(contents);
+	return `${type}-${hash.digest('base64')}`;
+});
+
+const mapOverAttrsAsync = async (lambda, set) => {
+	const set_copy = { ...set };
+
+	for (const attrName in set) {
+		set_copy[attrName] = await lambda(attrName, set_copy[attrName]);
+	}
+
+	return set_copy;
+};
```

Comment:
This one liner should get the job done. Also is this method necessary?
```suggestion
const mapOverAttrsAsync = async (lambda, set) => {
         return Object.fromEntries(Object.entries(set).map(([k, v]) => [k, await lambda(v)]));
};
```

Reply:
I didn't know about these `Object`' methods, they are useful indeed, thanks a lot ! I have removed `mapOverAttrsAsync` as it wasn't necessary.